### PR TITLE
Refactor controllers to use shared clients, listers, and informers

### PR DESCRIPTION
**Pull request by coreydaley**

### CODE REVIEW

**Comment by coreydaley** [comment]
/assign @dmage @ricardomaraschini @adambkaplan 

This PR also includes the import sorting from #481 so you can ignore that.
I know the tests aren't all passing yet, but it seems like those are mostly flakes, except for the unit tests which I am currently investigating.

File: pkg/client/generator.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+	regopinformers "github.com/openshift/client-go/imageregistry/informers/externalversions"
+	routeset "github.com/openshift/client-go/route/clientset/versioned"
+	routesetv1 "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+	routeinformers "github.com/openshift/client-go/route/informers/externalversions"
+
+	"github.com/openshift/cluster-image-registry-operator/defaults"
+	"github.com/openshift/cluster-image-registry-operator/pkg/parameters"
+)
+
+type Generator struct {
```
Comment (dmage) [inline pkg/client/generator.go]: Please give it another name, it's terrible to have the same meaningless name in two packages.

**Pull request closed without merging**